lib/src/commit_builder.rs

```diff
@@ -28,10 +26,21 @@ pub struct CommitBuilder<'repo, 'settings> {
     rewrite_source: Option<Commit>,
 }
 
+#[cfg(test)]
 /// TODO: This function is still used in some tests, but shouldn't be. Its
-/// output is not reproducible. Its replacement is available in a follow-up commit.
+/// output is not reproducible.
 pub fn new_change_id_deprecated() -> ChangeId {
-    ChangeId::from_bytes(Uuid::new_v4().as_bytes())
+    ChangeId::from_bytes(uuid::Uuid::new_v4().as_bytes())
+}
+
+pub fn new_change_id(settings: &UserSettings) -> ChangeId {
+    let random_bytes = settings.random_gen();
+    // The `uuid` crate is used merely to specify the number of random bytes (16)
+    ChangeId::from_bytes(
+        uuid::Builder::from_random_bytes(random_bytes)
+            .into_uuid()
+            .as_bytes(),
```

Comment:
I would have expected that we would still use UUID v4 if `debug.randomness-seed` was not set. Isn't there some reason that UUIDs are not just randomly generated bytes?

Reply:
Not as far as I can tell. According to https://docs.rs/uuid/latest/uuid/struct.Builder.html#method.from_random_bytes, this is what a V4 UUID is.  I looked at the source, and couldn't find anything happening with these bytes.

Reply:
Huh, you seem to be right. Then I think we should even get rid of the dependency on the UUID crate (unless we use it elsewhere). But that doesn't have to be in this PR.